extendeddatacrossword.go

```diff
@@ -229,11 +228,26 @@ func (eds *ExtendedDataSquare) prerepairSanityCheck(rowRoots [][]byte, columnRoo
 	for i := uint(0); i < eds.width; i++ {
 		rowMask := mask.RowView(int(i))
 		columnMask := mask.ColView(int(i))
-		if (vecIsTrue(rowMask) && !bytes.Equal(rowRoots[i], eds.RowRoots()[i])) || (vecIsTrue(columnMask) && !bytes.Equal(columnRoots[i], eds.ColumnRoots()[i])) {
-			return errors.New("bad roots input")
+		rowMaskIsVec := vecIsTrue(rowMask)
+		columnMaskIsVec := vecIsTrue(columnMask)
+
+		// if there's no missing data in the this row
+		if !checkForNil(eds.Row(i)) {
```

Comment:
What do you think of renaming the function to `noMissingData` and removing the `!`?

Reply:
that is definitely easier to read, thanks! resolved https://github.com/lazyledger/rsmt2d/pull/18/commits/193de535108c7547b5263fecec7716545bdde58f